spring-integration-smb/src/test/java/org/springframework/integration/smb/SmbTestSupport.java

```diff
@@ -28,35 +31,46 @@
 import jcifs.smb.SmbFile;
 
 /**
- * Provides a connection to an external SMB Server for test cases.
- *
- * The constants need to be updated with the 'real' server settings for testing.
+ * Provides a connection to a Testcontainers-driven SMB Server for test cases.
  *
  * @author Gregory Bragg
+ * @author Artem Vozhdayenko
  *
  * @since 6.0
  */
 
-@Disabled("Actual SMB share must be configured in class [SmbTestSupport].")
+@Testcontainers(disabledWithoutDocker = true)
 public class SmbTestSupport extends RemoteFileTestSupport {
 
-	public static final String HOST = "localhost";
+	public static final String HOST = "127.0.0.1";
 
-	public static final String SHARE_AND_DIR = "smb-share/";
+	public static final String SHARE_AND_DIR = "smb-share";
 
 	public static final String USERNAME = "sambaguest";
 
 	public static final String PASSWORD = "sambaguest";
 
+	private static final String INNER_SHARE_DIR = "/tmp";
+
+	// Configuration details can be found at https://hub.docker.com/r/dperson/samba
+	private static final GenericContainer<?> SMB_CONTAINER = new GenericContainer<>("dperson/samba:latest")
```

Comment:
The `latest` is not OK. Seems for me most of the images are resolved against snapshots.
So, better to use an explicit version.

Reply:
Agree, however this particular image does not have version tags (https://github.com/dperson/samba/issues/394).
In theory we could use sha256 digests, but I think this is not a good solution as well as it will be dependent on a concrete CPU architecture (if my understanding is correct).
Maybe you have any ideas?

Reply:
I don't know. Maybe we need to use some other vendor image, e.g. https://github.com/crazy-max/docker-samba ?..

Reply:
It, makes sense, I will try to change the code to be able to use another one.